modules/internal/OwnedObject.chpl

```diff
@@ -404,6 +404,12 @@ module OwnedObject {
     }
   }
 
+  @chpldoc.nodoc
+  proc _owned.deserialize(reader, ref deserializer) throws {
+    var tmp = this.chpl_p! : borrowed class;
+    reader.read(tmp);
```

Comment:
It looks like this reads into `tmp` and then drops the result on the floor. That doesn't seem right.

Reply:
``tmp`` is just the inner _owned pointer cast to 'borrowed', so it's still referring to the actual data and the 'read' isn't being lost.

Reply:
ah, right, thanks for correcting my understanding